datadog/api/http_client.py

```diff
@@ -94,6 +94,8 @@ def request(cls, method, url, headers, params, data, timeout, proxies, verify, m
                 u"Datadog's usage. We recommand upgrading it ('pip install -U requests')."
                 u"If you need help or have any question, please contact support@datadoghq.com"
             )
+        finally:
```

Comment:
Although this is 100% fine, I think I would prefer if we could use `requests` context manager for sessions...

we can basically wrap the entire method with:

```python
with requests.Session() as s:
    http_adapter = requests.adapters.HTTPAdapter(max_retries=max_retries)
    s.mount('https://', http_adapter)
    
    # REST OF METHOD...
```

If you don't want to update that, just let me know as this is blatant nitpicking, and we can merge this as is!

Reply:
updated

Reply:
let me know what you think